### ISSUE TITLE
Console and logging

### ISSUE THREAD

**Issue by Zardoz89** [trillek-team/tec#48]
A quake like console would be nice. Not only would show a log of what the engine is doing (very useful!), But would have a simple command interpreter to execute a few commands or execute LUA scripts; and list/view/edit engine config variables (like full-screen mode, key-bindings, etc..) exactly like what does Quake or Doom consoles.

**Comment by Zardoz89** [comment]
I actually have a [fork](https://github.com/Zardoz89/tec/tree/console) where I implemented a prototype of console that is hides/shows with ESC key and have bundle a little command interpreter, with a few embed commands (cmdlist, clear, echo, help) and a mechanism to add new commands (on main.cpp adds "exit" command)

I don't checked if we have a LUA, or what script language we would use, integrated on TEC, but I think that the console input should be redirected to the LUA interpreter so could execute LUA commands and scripts, and define these few embed commands on the script language.

Apart, I integrated [SpdLog](https://github.com/gabime/spdlog) on TEC, doing logging output to stdout and to the console. spdlog is a logger that is header only, lightweight, fast, multi-theread safe and have a async mode. Works very nice, and have a mechanism to register many loggers on a global storage (a multiton) that allow to grab it with a  `spdlog::get("logger_name")`. I don't think that we need multiple loggers, but this logger registry allows us to use it on every place of our codebase. The only requisite is that is necessary to define the logger and load tec::Console (this is safe) before anything.
I would like to begin to replace all our `std::cXXX` and `printf`s with the apropiate calls to the logger, if you agree with the usage of spdlog.
